src/components/structures/HomePage.tsx

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2020 New Vector Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import * as React from "react";
+
+import AutoHideScrollbar from './AutoHideScrollbar';
+import { getHomePageUrl } from "../../utils/pages";
+import { _t } from "../../languageHandler";
+import SdkConfig from "../../SdkConfig";
+import * as sdk from "../../index";
+import dis from "../../dispatcher";
+
+
+const onClickSendDm = () => dis.dispatch({action: 'view_create_chat'});
+const onClickExplore = () => dis.dispatch({action: 'view_room_directory'});
+const onClickNewRoom = () => dis.dispatch({action: 'view_create_room'});
+
+const HomePage = () => {
+    const config = SdkConfig.get();
+    const pageUrl = getHomePageUrl(config);
+
+    if (pageUrl) {
+        const EmbeddedPage = sdk.getComponent('structures.EmbeddedPage');
+        return <EmbeddedPage className="mx_HomePage" url={pageUrl} scrollbar={true} />;
+    }
+
+    const brandingConfig = config.branding;
```

Comment:
Aha, interesting, I guess I thought you might do this as the default content an embedded page rather than a native component, similar to https://github.com/vector-im/riot-web/blob/develop/res/welcome.html for the welcome page, but now I realise I never wrote that thought down anywhere. 😅 

With this approach, it's easier to build and maintain the default home page, but a bit harder to create a custom one since you can't really reuse this component and so would need to start from scratch with custom HTML.

Anyway, this seems fine for now, especially given our desire to land something today for the upcoming RC. If we end up with people trying to tweak the page and sad that they have start over, we can refactor to extract it out as an external page.

Reply:
It'd be harder to hook that into dialog actions. These actions do not have their own URLs.